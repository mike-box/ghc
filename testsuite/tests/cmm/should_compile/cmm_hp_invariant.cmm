#include "Cmm.h"

violate_hp_invariant (W_ counter)
{
  W_ x1, x2;
  x1 = Hp + 8;
  x2 = W_[x1+8];
  return (x2);
}
