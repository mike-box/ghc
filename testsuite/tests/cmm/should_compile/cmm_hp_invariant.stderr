ghc: panic! (the 'impossible' happened)
  GHC version 9.3.20211112:
	Cmm Hp invariant violated. The NCG backend currently assumes that
we never use Hp to read from memory. Nor any values which alias to Hp.
If you are using hand-written Cmm please adjust your code accordingly.
If you are not please report this as a GHC bug.


Please report this as a GHC bug:  https://www.haskell.org/ghc/reportabug

